packages/deployments/contracts/contracts/core/connext/facets/ProposedOwnableFacet.sol

```diff
@@ -245,12 +264,61 @@ contract ProposedOwnableFacet is BaseConnextFacet, IProposedOwnable {
     _setOwner(s._proposed);
   }
 
-  function pause() public onlyOwner {
+  /**
+   * @notice Use to assign an address Router role
+   * Address with Router has permission to add new router
+   * and assign addresses as Role.Router
+   * Can only be called by Owner or Role.Router
+   * @dev requested address will be whitelisted as Role.Router under roles mapping
+   * @param _router - The address to be assigned as Role.Router under roles
+   */
+  function assignRoleRouter(address _router) public onlyOwnerOrRouter {
+    // Use contract as source of truth
+    // Will fail if candidate is already added
+    if (s.roles[_router] == Role.Router) revert ProposedOwnableFacet__assignRoleRouter_alreadyAdded();
+
+    s.roles[_router] = Role.Router;
+    emit AssignRouterRole(_router);
+  }
+
+  /**
+   * @notice Use to assign an address Watcher role
+   * Address with Watcher role has permission to pause
+   * Can only be called by Owner or Role.Admin
+   * @dev requested address will be whitelisted as Role.Watcher under roles mapping
+   * @param _watcher - The address to be assigned as Role.Watcher under roles
+   */
+  function assignRoleWatcher(address _watcher) public onlyOwnerOrAdmin {
+    // Use contract as source of truth
+    // Will fail if candidate is already added
+    if (s.roles[_watcher] == Role.Watcher) revert ProposedOwnableFacet__assignRoleWatcher_alreadyAdded();
+
+    s.roles[_watcher] = Role.Watcher;
+    emit AssignWatcherRole(_watcher);
+  }
+
+  /**
+   * @notice Use to assign an address Admin role
+   * Address with Admin role has permission to all else of Router & Watcher role
+   * Can only be called by Owner or Role.Admin
+   * @dev requested address will be whitelisted as Role.Admin under roles mapping
+   * @param _admin - The address to beassigned as Role.Admin under roles
+   */
+  function assignRoleAdmin(address _admin) public onlyOwnerOrAdmin {
```

Comment:
I feel like we should have role acceptance methods (i.e. use the pending -> accepted pattern).

If not, at the very least, we probs wanna sanity check that the input address isn't `address(0)` at least, since having s.roles[0x0] = Roles.Admin could have bad consequences lol

Reply:
will add sanity check for addressZero